### DISCUSSION TITLE
Python 3.10

### DISCUSSION THREAD

**Discussion by WaylonWalker** [kedro-org/kedro#940] [Idea]
I was just looking through the release notes for 3.10 and saw they have done some amazing things to improve error messages.  That alone seems worth the price of admission.  Is there any way that we can tag kedro as being available in 3.10?

**Comment by datajoely** [comment]
So keen to do this - however, the reason we don't have 3.9 support yet is because we would have to issue a breaking change and as such as will be made available in the next major version 0.18.0

As such, support for Python 3.10 will be made available in that version of Kedro in best case scenario. We're still aiming to issue 0.18.0 Q4 of this year, but it may fall into early 2022.

**Reply by WaylonWalker** [reply]
Is the 3.9 issue within the core library or within optional dependencies?

**Reply by datajoely** [reply]
I'm hearing from the team it is to do with one of our pandas.* dataset dependencies rather than something Python itself - but because it will break people's workflows we can't release it without issuing a major version.

**Comment by datajoely** [comment]
Hi @WaylonWalker 3.9 and 3.10 support are now out in Kedro 0.18.0!

**Reply by WaylonWalker** [reply]
This is fantastic, thanks for sharing @datajoely !
